Rideau/SwiftUI/SwiftUISuppports.swift

```diff
@@ -0,0 +1,163 @@
+
+      var c = configuration
+
+      c.snapPoints.insert(.hidden)
+
+      self.rideauView = .init(frame: .zero, configuration: c)
+
+      self.backgroundColor = backdropColor
+
+      super.init(nibName: nil, bundle: nil)
+
+      self.backgroundView.backgroundColor = .clear
+
+      do {
+
+        if usesDismissalPanGestureOnBackdropView {
+
+          let pan = UIPanGestureRecognizer()
+
+          backgroundView.addGestureRecognizer(pan)
+
+          rideauView.register(other: pan)
+
+        }
+
+      }
+
+    }
+
+    @available(*, unavailable)
+    required init?(
+      coder aDecoder: NSCoder
+    ) {
+      fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Functions
+
+    override func viewDidLoad() {
+      super.viewDidLoad()
+
+      do {
+
+        if hidesByBackgroundTouch {
+          let tap = UITapGestureRecognizer(target: self, action: #selector(didTapBackdropView))
+          backgroundView.addGestureRecognizer(tap)
+        }
+
+        view.addSubview(backgroundView)
+
+        view.backgroundColor = .clear
+
+        backgroundView.frame = view.bounds
+        backgroundView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+
+        view.addSubview(rideauView)
+        rideauView.frame = view.bounds
+        rideauView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+
+        // To create resolveConfiguration
+        view.layoutIfNeeded()
+
+        set(bodyViewController: bodyViewController, to: rideauView, resizingOption: resizingOption)
+
+        view.layoutIfNeeded()
+      }
+
+      rideauView.handlers.willMoveTo = { [weak self] point in
+
+        guard let self else { return }
+
+        guard point == .hidden else {
+
+          UIViewPropertyAnimator(duration: 0.6, dampingRatio: 1) {
+            self.backgroundView.backgroundColor = self.backgroundColor
+          }
+          .startAnimation()
+
+          return
+        }
+
+        UIViewPropertyAnimator(duration: 0.6, dampingRatio: 1) {
+          self.backgroundView.backgroundColor = .clear
+        }
+        .startAnimation()
+
+      }
+
+      rideauView.handlers.didMoveTo = { [weak self] point in
+
+        guard let self = self else { return }
+
+        guard point == .hidden else {
+          return
+        }
+
+        self.onWillDismiss()
+        self.onDidDismiss()
+
+      }
+
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+      super.viewDidAppear(animated)
+      onViewDidAppear(self)
+    }
+
+    func set(
+      bodyViewController: UIViewController,
+      to rideauView: RideauView,
+      resizingOption: RideauContentContainerView.ResizingOption
+    ) {
+      bodyViewController.willMove(toParent: self)
+      addChild(bodyViewController)
+      rideauView.containerView.set(bodyView: bodyViewController.view, resizingOption: resizingOption)
+    }
+
+    @objc private dynamic func didTapBackdropView(gesture: UITapGestureRecognizer) {
```

Comment:
Do we need it to by dynamic?

Reply:
ah it's kind of classic way to support objc dynamics. in some swift version, swift compiler deleted those symbols in optimization. we could solve that by adding dynamic. 
but seems nowadays we don't need to do that anymore.